mono/metadata/object.c

```diff
@@ -7059,16 +7060,7 @@ mono_ldstr_utf8 (MonoImage *image, guint32 idx, MonoError *error)
 char *
 mono_string_to_utf8 (MonoString *s)
 {
-	MONO_REQ_GC_UNSAFE_MODE;
-
-	MonoError error;
-	char *result = mono_string_to_utf8_checked (s, &error);
-	
-	if (!is_ok (&error)) {
-		mono_error_cleanup (&error);
-		return NULL;
-	}
-	return result;
+	return mono_string_to_utf8_checked (s, NULL);
```

Comment:
the usual pattern for `MONO_RT_EXTERNAL_ONLY` functions that doesn't take a `MonoError*` is to write something like:

```c
char *
mono_string_to_utf8 (MonoString *s)
{
  MonoError error;
  char *result = mono_string_to_utf8_checked (s, error);
  mono_error_cleanup (&error); // FIXME: export a better API that doesn't swallow the error
  return result;
}
```
and write the `_checked` version with a mandatory non-`NULL` `MonoError*` argument.